src/client/Microsoft.Identity.Client/OAuth2/MsalTokenResponse.cs

```diff
@@ -176,7 +180,11 @@ internal static MsalTokenResponse CreateFromAndroidBrokerResponse(string jsonRes
                 ExtendedExpiresIn = DateTimeHelpers.GetDurationFromNowInSeconds(authResult[BrokerResponseConst.ExtendedExpiresOn].ToString()),
                 ClientInfo = authResult[BrokerResponseConst.ClientInfo].ToString(),
                 TokenType = authResult[BrokerResponseConst.TokenType]?.ToString() ?? "Bearer",
-                TokenSource = TokenSource.Broker
+                TokenSource = TokenSource.Broker,
+                AuthorityUrl = authResult[BrokerResponseConst.Authority].ToString(),
```

Comment:
Do you need to add the null protection `.?`  here as well?

Reply:
It should be present when successful. Added for safety.